hack/schemas/main_test.go

```diff
@@ -73,3 +77,92 @@ func TestGenerators(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateJsonSchemas(t *testing.T) {
+	root := "../.."
+
+	for _, v := range schema.SchemaVersions {
+		apiVersion := strings.TrimPrefix(v.APIVersion, "skaffold/")
+		t.Run(apiVersion, func(t *testing.T) {
+			yamlMapConfig, err := toMapWithYamlKeys(v.Factory())
+			if err != nil {
+				t.Fatalf("cannot convert to config with yaml names")
+			}
+			document := gojsonschema.NewGoLoader(yamlMapConfig)
+
+			rawSchema := loadSchemaForAPIVersion(t, root, apiVersion)
+			jsonSchema := gojsonschema.NewStringLoader(rawSchema)
+
+			result, err := gojsonschema.Validate(jsonSchema, document)
+			if err != nil {
+				t.Fatalf("schema validation failed: %s", err)
+			}
+
+			if !result.Valid() {
+				for _, desc := range result.Errors() {
+					t.Error(desc)
+				}
+			}
+		})
+	}
+}
+
+func loadSchemaForAPIVersion(t *testing.T, root, apiVersion string) string {
+	t.Helper()
+	schemaPath := fmt.Sprintf("%s/docs/content/en/schemas/%s.json", root, apiVersion)
+	if _, err := os.Stat(schemaPath); err == nil {
+		if content, err := ioutil.ReadFile(schemaPath); err != nil {
+			t.Errorf("unable to read existing schema for version %s", apiVersion)
+		} else {
+			return string(content)
+		}
+	} else if os.IsNotExist(err) {
+		t.Skip("could not find schema at", schemaPath)
+	}
+	return ""
```

Comment:
if this gets returned and passed to `gojsonschema.NewStringLoader()` and then validated, it will pass right? do we consider this a success case? or should we just error out